src/test/java/calculator/StringCalculator.java

```diff
@@ -0,0 +1,43 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.thymeleaf.util.StringUtils;
+
+public class StringCalculator {
+
+    public static final String DELIMITER = ",|:";
+    public static final String SEPARATOR = "|";
+    public static final Pattern pattern = Pattern.compile("//(.)\n(.*)");
+    public static final int ZERO = 0;
+    public static final int ONE = 1;
+    public static final int TWO = 2;
+    public static final String NEGATIVE_EXCEPTION_MESSAGE = "음수는 계산할 수 없습니다";
+
+    public int calculate(final String stringNumber) {
+        if (StringUtils.isEmpty(stringNumber)) {
+            return ZERO;
+        }
+        final Matcher matcher = pattern.matcher(stringNumber);
+        if (matcher.find()) {
+            final String customDelimiter = matcher.group(ONE);
+            final String targetNumber = matcher.group(TWO);
+            return calculate(targetNumber, DELIMITER + SEPARATOR + customDelimiter);
+        }
+        return calculate(stringNumber, DELIMITER);
```

Comment:
```suggestion
        }
        
        final Matcher matcher = pattern.matcher(stringNumber);
        if (matcher.find()) {
            final String customDelimiter = matcher.group(ONE);
            final String targetNumber = matcher.group(TWO);
            return calculate(targetNumber, DELIMITER + SEPARATOR + customDelimiter);
        }
        return calculate(stringNumber, DELIMITER);
```
공백 라인을 잘 활용한다면 전체적인 흐름을 파악하기 쉬워질거에요. 
문단 단위로 공백을 추가해보세요 :)

Reply:
공감합니다. 공백 라인을 활용해볼게요.
평소에도 궁금했던건데 리뷰어님(팀)은 공백을 나누는 기준이 따로 정해져 있으신가요?
팀에서 공백 라인 컨벤션을 제시하기가 애매하더라고요. (intellij가 공백까지 자동으로 넣어주지는 않아서..)
예를들어 위 코드에서 마지막 return문 위는 공백인가? 아닌가? 사람마다 다를 수 있다고 생각해요.

Reply:
저희 팀은 공백을 나누는 기준은 따로 정해져있지는 않아요. 
코드 리뷰 진행시 문맥이 안맞거나 공백 활용을 잘못한 부분에 대해서 코멘트를 남기다 보니 자연스럽게 익숙해지더라구요 ㅎㅎ
개인적으로는 return 시 공백을 추가하는 편입니다 !